rust/flatbuffers/src/array.rs

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::marker::PhantomData;
+use std::fmt::{Debug, Formatter, Result};
+use std::mem::size_of;
+use std::slice::from_raw_parts;
+use crate::vector::{VectorIter, SafeSliceAccess};
+use crate::follow::Follow;
+
+#[derive(Copy, Clone)]
+pub struct Array<'a, T: 'a, const N: usize>(&'a [u8], PhantomData<T>);
```

Comment:
Just curious, why not pass the N into the `&'a [u8]` array itself?

Like this

```rust
pub struct Array<'a, T: 'a, const N: usize>(&'a [u8, N], PhantomData<T>);
```

Note: I haven't tested it this way. Just curious about the difference between doing it that way and the current way with into.

Reply:
`&[u8]` is actually a slice and notice it's type is different from the array's items `T` vs `u8` so it will require something like `&'a [u8; N * core::mem::size_of::<T>()]` which I'm not sure Rust supports yet.

To make it more sound we can add an assert  that verifies the slice is bigger than `N * core::mem::size_of::<T>()`

Reply:
yea if we can make the number of u8s static, that'd be nice :)

Reply:
Yeah, it requires complex generic expressions in constants which has an RFC already but is far from being stable :(